sphinx_doc/howtos/tutorials/docs/navigation2_on_real_turtlebot3.rst

```diff
@@ -0,0 +1,161 @@
+- ``export TURTLEBOT3_MODEL=waffle``
+
+1- Launch Turtlebot 3 Robot State Publisher
+-------------------------------------------
+
+Launch Turtlebot 3 robot state publisher,
+
+  ``ros2 launch turtlebot3_bringup turtlebot3_state_publisher.launch.py use_sim_time:=False``
+
+Make sure use_sim_time is set to **False**.
+
+2- Launch Navigation2
+---------------------
+
+You need to have a map of the environment where you want to Navigate Turtlebot 3. Required files:
+
+   - ``your-map.map``
+   - ``your-map.yaml``
+
+``<your_map>.yaml`` is the configuration file for the map we want to provide Navigation2.
+In this case, it has the map resolution value, threshold values for obstacles and free spaces, and a map file location.
+You need to make sure these values are correct. 
+More information about the map.yaml can be found `here <http://wiki.ros.org/map_server>`_.
+
+Launch Navigation 2. If you set autostart:=False, you need to click on the start button in RViz to initialize the nodes. 
+Make sure `use_sim time` is set to **False**, because we want to use the system time instead of the time simulation time from Gazebo.
+
+``ros2 launch nav2_bringup nav2_bringup_launch.py use_sim_time:=False autostart:=False map:=/path/to/your-map.yaml``
+
+Don't forget to change **/path/to/your-map.yaml** to the actual path to the your-map.yaml file.
+
+3-  Launch RVIZ
+---------------
+
+Launch RVIZ with a pre-defined configuration file.
+
+  ``ros2 run rviz2 rviz2 -d $(ros2 pkg prefix nav2_bringup)/share/nav2_bringup/launch/nav2_default_view.rviz``
+
+Now, you should see a shadow of Turtlebot 3 robot model in the center of the plot in Rviz.
+Click on the Start button (Bottom Left) if you set the auto_start parameter to false.
+
+.. image:: images/Navigation2_on_real_Turtlebot3/rviz_after_launch_view.png
+    :height: 720px
+    :width: 1024px
+    :alt: RViz after launch, auto_start = False
+
+Now, the map should appear in RViz.
+
+.. image:: images/Navigation2_on_real_Turtlebot3/rviz_slam_map_view.png
+    :height: 720px
+    :width: 1024px
+    :alt: A map generated by using SLAM in RViz
+
+4- Initialize the Location of Turtlebot 3
+-----------------------------------------
+
+First, find where the robot is on the map. Check where your robot is in the room.
+
+Set the pose of the robot in RViz. 
+Click on the 2D Pose Estimate button and point the location of the robot on the map. 
+The direction of the green arrow is the orientation of Turtlebot.
+
+.. image:: images/Navigation2_on_real_Turtlebot3/rviz_set_initial_pose.png
+    :height: 720px
+    :width: 1024px
+    :alt: Set initial pose in RViz
+
+Now, the 3D model of Turtlebot should move to that location. 
+A small error in the estimated location is tolerable.
+
+5-  Send a Goal Pose
+--------------------
+
+Pick a target location for Turtlebot on the map. 
+You can send Turtlebot 3 a goal position and a goal orientation by using the **Navigation2 Goal** and the **GoalTool** buttons.
```

Comment:
_or_ the GoalTool

Reply:
Does GoalTool still work?

Reply:
Yes. I shipped an updated topic to rviz2 a while back

Reply:
Actually, Aren't the Navigation2 Goal and the Goal Tool are the same thing. I meant the RViz Set Goal tool (2D Goal Pose). 

It should be **Navigation2 Goal** or the **2D Goal Pose** buttons

Reply:
Are you sure its not just your rviz config? The default rviz config works. Its [in the eloquent guide](https://index.ros.org/doc/ros2/Releases/Release-Eloquent-Elusor/) because I changed it.

Reply:
Apparently, the default Rviz Goal tool publishes a topic named /goal and there is no subscriber for the /goal topic. There is also another topic called goal_pose and there is a subscriber for that topic. Publishing to the goal_pose manually using ros2 topic doesn't do anything either..

Reply:
I don't know maybe my rviz is old.

Reply:
but publishing to goal_pose

``ros2 topic pub /goal_pose geometry_msgs/msg/PoseStamped "{header: {stamp: {sec: 0}, frame_id: 'map'}, pose: {position: {x: 0.22, y: 0.54, z: 0.0}, orientation: {w: 1.0}}}" `` 

doesn't move the robot.

Reply:
did you check the callback in BT nav that its getting the message and that BT nav is subscribed to that topic?

Reply:
Did some digging the last 20 minutes, you nearly gave me a heart attack that I introduced a bug into rviz a while back :rofl: 

Ok. So. This is what's up. 

https://github.com/ros2/rviz/pull/455/files#diff-ff71e74f1afc1e3d1004ead9979c1ac3R66

That line in that commit changes the **default** rviz file that ships with the repo when installed to have the topic `/goal_pose` for the `SetGoal` tool. The setgoal tool is a masked registration of the goal pose tool, 

https://github.com/ros2/rviz/pull/455/files#diff-b457699f4d5c67c16326728124de9712R51

in the same commit. As you can see though the line I highlighted, when you **dont** override with the default rviz file, it will be exposed as `goal`. 

So this leads me to believe you just have a bad config file, or you're working on the **dashing** install of rviz, and not the master branch updated anytime since September 4th.

Reply:
I assumed that it would print something like  ``[bt_navigator]: Begin navigating from current location to (0.22, 0.54)`` in the terminal. It didn't print anything.

Has this feature been tested since Dashing?

Reply:
No idea. This is a good reason to only have **one** goal pose tool and delete ours.

Reply:
https://github.com/ros-planning/navigation2/blob/master/nav2_bringup/bringup/rviz/nav2_namespaced_view.rviz#L331

Our rviz files don't contain the `goal_pose` override. But I also just PRed against rviz2 to also just change the default https://github.com/ros2/rviz/pull/491

Reply:
I build ROS2 master every 10-15 days. I fresh built Nav2 master to test this, so I don't think the rviz conf file was corrupted.

Like you mentioned in our rviz we don't override, so it publishes to the default topic /goal instead of /goal_pose.

I was using this rviz file.

https://github.com/ros-planning/navigation2/blob/0154951f9cbeab9b5272f36ed02d65b25fdda0f4/nav2_bringup/bringup/rviz/nav2_default_view.rviz

your rviz2 PR should fix this issue.

> you nearly gave me a heart attack that I introduced a bug into rviz a while back

Haha.. who knew a documentation PR would cause this much headache :)

Now, I need to figure out why publishing to /goal_pose via cli didn't move the robot.

Reply:
Again... not really an issue, its the nav2 rviz configs that are broken...

Reply:
I just submitted a PR for this issue please take a look at it. #1434